hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/NodeExecutor.java

```diff
@@ -47,6 +47,7 @@ public void launchTask(final ExecutorDriver driver, final TaskInfo taskInfo) {
     executorInfo = taskInfo.getExecutor();
     task = new Task(taskInfo);
     startProcess(driver, task);
+    new Thread(new TaskSupervisor(driver, task)).start();
     driver.sendStatusUpdate(TaskStatus.newBuilder()
         .setTaskId(taskInfo.getTaskId())
```

Comment:
I'm reverting this logic. if we decide to overhaul the while loop HA pattern in the future, something similar should probably go in AbstractNodeExecutor to watch the process and update the task status accordingly (and re-run the task of course).